### PR TITLE
[standalone] Start ZooKeeper and Kafka Services

### PR DESCRIPTION
**Pull request by eolivelli**
## Summary

This is a follow up work in order to allow running Venice Standalone more easily.

Contents:
- Allow to start the ZooKeeper server
- Allow to start the Kafka Broker
- Allow to select the Services to start

Basic command:
`./gradlew :services:venice-standalone:run`

Selection of services:
`./gradlew :services:venice-standalone:run -Dservices=zookeeper,kafka,controller,server,router`

Notes:
- ZooKeeper is started using Apache Curator Testing utilities (it would be great to use the official EmbeddedZooKeeper API, but it is not available in 3.5.x)
- We have to forcibly delete the Kafka Broker ephemeral znode on startup, because when the ZK service is down ephemeral nodes are not deleted and so Kafka broker is not able to restart

## How was this PR tested?

Only manual testing

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by adamxchen** [comment]
> can anyone restart the failed job please ?

Restarted!

File: services/venice-standalone/zookeeperData/version-2/snapshot.0
Comment (adamxchen) [inline services/venice-standalone/zookeeperData/version-2/snapshot.0:1]: what's this file about? Does it have to be present in the commit? I feel that's a file generated during runtime.

Comment (eolivelli) [reply services/venice-standalone/zookeeperData/version-2/snapshot.0:1]: it is in .gitignore
maybe I did some errors while committing.

removed

File: services/venice-standalone/src/main/java/com/linkedin/venice/standalone/VeniceStandalone.java [hunk 1]
```diff
@@ -35,4 +68,72 @@ public static void main(String[] args) throws Exception {
...
+      try (Reader reader = new FileReader(zookeeperConfig)) {
+        properties.load(reader);
+      }
+      boolean enabled = Boolean.parseBoolean(properties.getProperty("enabled", "false"));
+      if (enabled) {
+        String dataDirectory = properties.getProperty("dataDir", "zookeeperData");
+        int port = Integer.parseInt(properties.getProperty("port", "2181"));
+        LOGGER.info("Starting ZooKeeper on port {}, dataDir is {}", port, dataDirectory);
+        InstanceSpec spec = new InstanceSpec(new File(dataDirectory), port, -1, -1, false, 1);
+        TestingServer zkServer = new TestingServer(spec, false);
```
Comment (adamxchen) [inline services/venice-standalone/src/main/java/com/linkedin/venice/standalone/VeniceStandalone.java:93]: I've heard curator is easy to use but I wonder whether we can just use the [ZKServerWrapper](https://github.com/linkedin/venice/blob/9653d18ca78d69f972f3dc5b2fa3a8dde0c11966/internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/integration/utils/ZkServerWrapper.java#L29) we have been using in the integ tests.

My concern is having `curator` would introducing duplicate dependencies on Zk and might increase some overhead/confusion in the future

Comment (eolivelli) [reply services/venice-standalone/src/main/java/com/linkedin/venice/standalone/VeniceStandalone.java:93]: This is only for Standalone.

A better alternative would be to upgrade to ZooKeeper 3.7+

Comment (FelixGV) [reply services/venice-standalone/src/main/java/com/linkedin/venice/standalone/VeniceStandalone.java:93]: Am open to considering the ZK upgrade. I would just like to get some input from our ZK team about whether upgrading the client to 3.7 ahead of the ZK server could cause any issues.

I'm curious though, why would upgrading to 3.7 help here? Is there better support for spinning it up standalone in that version?

**Comment by adamxchen** [comment]
@eolivelli  I've restarted the workflow. Can you plz resolve the conversation (and the conflict files too) so we can merge this?

**Comment by adamxchen** [comment]
Hey @eolivelli , do you have any bandwidth to iterate the PR and resolve the conflicts? We'd like to keep the open PR list short so folks can focus on high-priority ones and get changes into production ASAP.

**Comment by adamxchen** [comment]
I'm going to close this PR since there's no traction for some time. However, please feel free to re-open when you get time to work on it.

**Pull request closed without merging**